nifi-nar-bundles/nifi-smb-bundle/nifi-smb-processors/src/main/java/org/apache/nifi/processors/smb/ListSmb.java

```diff
@@ -0,0 +1,304 @@
+            .displayName("Minimum file age in milliseconds")
+            .name("min-age")
+            .description(
+                    "Any file younger the the given value will be omitted. Ideally this value should be greater then"
+                            + "the amount of time needed to perform a list.")
+            .required(true)
+            .addValidator(INTEGER_VALIDATOR)
+            .addValidator(NON_NEGATIVE_INTEGER_VALIDATOR)
+            .defaultValue("5000")
+            .build();
+
+    public static final PropertyDescriptor SMB_LISTING_STRATEGY = new PropertyDescriptor.Builder()
+            .fromPropertyDescriptor(LISTING_STRATEGY)
+            .allowableValues(BY_ENTITIES, NO_TRACKING, BY_TIMESTAMPS)
+            .build();
+
+    public static final PropertyDescriptor SMB_CONNECTION_POOL_SERVICE = new Builder()
+            .name("smb-connection-pool-service")
+            .displayName("SMB Connection Pool Service")
+            .description("Specifies the SMB Connection Pool to use for creating SMB connections.")
+            .required(true)
+            .identifiesControllerService(SmbConnectionPoolService.class)
+            .build();
+
+    public static final PropertyDescriptor SKIP_FILES_WITH_SUFFIX = new Builder()
+            .name("file-name-suffix-filter")
+            .displayName("File name suffix filter")
+            .description("Files ends with the given suffix will be omitted. This is handy when writing large data into "
+                    + "temporary files and then moved to a final one. Please be advised that writing data into files "
+                    + "first is highly recommended when using Entity Tracking or Timestamp based listing strategies.")
+            .required(false)
+            .addValidator(NON_EMPTY_VALIDATOR)
+            .addValidator(new MustNotContainDirectorySeparatorsValidator())
+            .build();
+    public static final PropertyDescriptor SHARE = new PropertyDescriptor.Builder()
+            .displayName("Share")
+            .name("share")
+            .description("The network share to which files should be listed from. This is the \"first folder\"" +
+                    "after the hostname: smb://hostname:port\\[share]\\dir1\\dir2")
+            .required(false)
+            .addValidator(NON_BLANK_VALIDATOR)
+            .build();
+
+    private static final List<PropertyDescriptor> PROPERTIES = unmodifiableList(asList(
+            AbstractListProcessor.TARGET_SYSTEM_TIMESTAMP_PRECISION,
+            AbstractListProcessor.RECORD_WRITER,
+            ListedEntityTracker.TRACKING_STATE_CACHE,
+            ListedEntityTracker.TRACKING_TIME_WINDOW,
+            ListedEntityTracker.INITIAL_LISTING_TARGET,
+            SMB_LISTING_STRATEGY,
+            SMB_CONNECTION_POOL_SERVICE,
+            DIRECTORY,
+            MINIMUM_AGE,
+            SKIP_FILES_WITH_SUFFIX,
+            SHARE
```

Comment:
Please order the properties from most important to least important.
Suggested order:
- SMB_LISTING_STRATEGY
- SMB_CONNECTION_POOL_SERVICE
- SHARE
- DIRECTORY
- AbstractListProcessor.RECORD_WRITER
- SKIP_FILES_WITH_SUFFIX
- MINIMUM_AGE
- AbstractListProcessor.TARGET_SYSTEM_TIMESTAMP_PRECISION
- ListedEntityTracker.TRACKING_STATE_CACHE
- ListedEntityTracker.TRACKING_TIME_WINDOW
- ListedEntityTracker.INITIAL_LISTING_TARGET